#167865032 Admin cancel trip

amkayondo opened the pull request:
#### What does this PR do?
Modifies cancel trip controller and trip models to enable admin to update the trip status to cancelled
#### Description of Task to be completed?
Admin can cancel a trip
#### How should this be manually tested?
send `PATCH` - `/trips/<:tripId>`
#### Any background context you want to provide?
`200` - OK
`404` - NOT FOUND
#### What are the relevant pivotal tracker stories?
[#167865032](https://www.pivotaltracker.com/story/show/167865032)
#### Screenshots (if appropriate)
![Screenshot (46)](https://user-images.githubusercontent.com/46356108/62953684-3c1e5e00-bdee-11e9-8200-ca510287b460.png)
![Screenshot (47)](https://user-images.githubusercontent.com/46356108/62953690-3f194e80-bdee-11e9-9d0a-e4fb2a182716.png)

amkayondo committed:
chore(API) Get trip by Id
*DEV Notes*
- modify controller to get trip by Id
- validate if the id is an integer
- modfiy trip tests

amkayondo committed:
chore (API): Modify update trip status by Admin

- Add Queries to db to update the trip status
- add more functionallity to controller
- Remove inline comments
- add eslint rules to eslintrc file

amkayondo committed:
chore(API): Modify trips
- add more functionallity to controller
- Remove inline comments
- add eslint rules to eslintrc file

Kasulejoseph commented:
Check your commit messages, they don't have a story ID. `[... #122222] `

Kasulejoseph commented on server/controllers/trips/createTrip.js:
```diff
@@ -1,41 +1,42 @@
+const createTrip = async (req, res) => {
+  try {
+  // const result = Joi.validate(req.body, tripSchema);
+  // if (result.error) {
```
You still have this commented code. Please get rid of it.

Kasulejoseph commented on server/controllers/users/signup.js:
```diff
@@ -6,39 +6,40 @@ import signUpSchema from '../../helpers/schema/signup';
+    res.header('Authorization', token);
+    await newUser.createNewUser(newData);
+    resPonse.successUser(res, 201, 'Account successfully created', token);
+  } catch (err){}
```
You're doing nothing with this catch error `err`, You may not see the effect on development but this may crush your API on production. Do something about it.

Kasulejoseph commented on server/models/trips.js:
```diff
@@ -19,16 +17,34 @@ const Trip = {
+    try {
+      const result = await db.selectItemById('trips', 'tripid', tripInput);
+      return result;
+    } catch (err) {}
```
Same here...

Kasulejoseph commented on server/models/users.js:
```diff
@@ -22,37 +19,37 @@ const User = {
-    ];
-    const cret = await db.addNewUser(userData);
-    return cret;
+    try {
```
Try to implement Javascript's OOP in this file, This will make it more modular.

amkayondo committed:
chore(API) Imprement feedback
- Add OOP to User Model
- Remove unneccessary catch and error sentments
- Add responses to catch errors
- Remove commented code
- Add validators to routes
[starts #167865032]

amkayondo committed:
chore(TESTS) Fix Travis crashes
- fix test 2tatus code from 201 to got 400
[starts #167865032]

amkayondo committed:
chore(API): Add OOP to trips
- Add classes and inherit methods

amkayondo committed:
chore(API): Implement all the feedback by LFA
- fix naming conventions as applied in th ADC
- Add response messages
- remove all commented code
- remove inline comments

Kasulejoseph commented on server/controllers/users/signup.js:
```diff
@@ -1,44 +1,41 @@
-    password,
-    xc,
-  };
+    const xc = false;
```
What is xc? provide more descriptive variable names.

amkayondo committed:
chore(API): change xc to isadmin
- Add more meaningful names for variables
